aws-wafv2-loggingconfiguration/aws-wafv2-loggingconfiguration.json

```diff
@@ -0,0 +1,258 @@
+{
+  "typeName": "AWS::WAFv2::LoggingConfiguration",
+  "description": "A WAFv2 Logging Configuration Resource Provider",
+  "sourceUrl": "https://github.com/advaj/aws-cloudformation-resource-providers-wafv2.git",
+  "definitions": {
+    "Filter": {
+      "type": "object",
+      "properties": {
+        "Behavior": {
+          "description": "How to handle logs that satisfy the filter's conditions and requirement. ",
+          "type": "string",
+          "enum": [
+            "KEEP",
+            "DROP"
+          ]
+        },
+        "Conditions": {
+          "description": "Match conditions for the filter.",
+          "type": "array",
+          "minItems": 1,
+          "items": {
+            "$ref": "#/definitions/Condition"
+          }
+        },
+        "Requirement": {
+          "description": "Logic to apply to the filtering conditions. You can specify that, in order to satisfy the filter, a log must match all conditions or must match at least one condition.",
+          "type": "string",
+          "enum": [
+            "MEETS_ALL",
+            "MEETS_ANY"
+          ]
+        }
+      },
+      "additionalProperties": false,
+      "required": [
+        "Behavior",
+        "Conditions",
+        "Requirement"
+      ]
+    },
+    "Condition": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "ActionCondition": {
+          "description": "A single action condition.",
+          "type": "object",
+          "additionalProperties": false,
+          "properties": {
+            "Action": {
+              "description": "Logic to apply to the filtering conditions. You can specify that, in order to satisfy the filter, a log must match all conditions or must match at least one condition.",
+              "type": "string",
+              "enum": [
+                "ALLOW",
+                "BLOCK",
+                "COUNT"
+              ]
+            }
+          },
+          "required": [
+            "Action"
+          ]
+        },
+        "LabelNameCondition": {
+          "description": "A single label name condition.",
+          "type": "object",
+          "additionalProperties": false,
+          "properties": {
+            "LabelName": {
+              "description": "The label name that a log record must contain in order to meet the condition. This must be a fully qualified label name. Fully qualified labels have a prefix, optional namespaces, and label name. The prefix identifies the rule group or web ACL context of the rule that added the label. ",
+              "type": "string"
+            }
+          },
+          "required": [
+            "LabelName"
+          ]
+        }
+      }
+    },
+    "FieldToMatch": {
+      "description": "A key-value pair to associate with a resource.",
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "JsonBody": {
```

Comment:
I think we want to model the JsonBody similar to what we already have : https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-wafv2/blob/main/aws-wafv2-rulegroup/aws-wafv2-rulegroup.json#L116-L175

For example the includedPath has a pattern restriction.

Reply:
We can discuss this offline as well, but a couple of thoughts: 
1. I have modelled this according to how the API is today - https://docs.aws.amazon.com/waf/latest/APIReference/API_PutLoggingConfiguration.html and https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/wafv2/model/JsonBody.html - hence, the structure for JSONBody. I feel that this is more intuitive for the end user. 
2. Regarding Regexes for attributes, I have shied away from using them because - if the regex changes and we have more combinations allowed then the CFN resource should not become yet another thing that we have to update/release. If that is not the case and you see that this regex is here to stay, I will incorporate it.

Reply:
As discussed, I have changed the Regex and have retained the JSONBody modelling.